Neo4j.Driver/Neo4j.Driver.Tests/Result/PeekingEnumeratorTests.cs

```diff
@@ -115,15 +115,32 @@ public void ShouldConsumeAllItemsAfterDiscard()
         public class DisposeMethod
         {
             [Fact]
-            public void ShouldNotDiscardAfterDispose()
+            public void ShouldDiscardAfterDispose()
             {
                 var list = new List<string> { "1", "2", "3" }.GetEnumerator();
                 var enumerator = new PeekingEnumerator<string>(list);
-                enumerator.MoveNext().Should().BeTrue();
-                enumerator.Consume();
+                enumerator.Dispose();
                 enumerator.MoveNext().Should().BeFalse();
                 enumerator.Peek().Should().BeNull();
-                enumerator.Position.Should().Be(3);
+                enumerator.Position.Should().Be(-1);
+            }
+
+            [Fact]
+            public void PeakShouldReturnDefaultAfterDispose()

```

Comment:
I agree. Peak should return default(T) instead of null in the code: `return CacheNext() ? _cached : null;`

Sorry, I missed that one..